Rating our skill feature

ram-nad opened the pull request:
:sparkles: Feature to rate our skill on scale of 1-5.

![Screenshot_20191001_193402](https://user-images.githubusercontent.com/44163356/65969760-f5122800-e482-11e9-8394-ab4ab79042ef.png)

aamay001 commented:
Very interesting feature. I will review and send back any change requests is needed. Thanks for your submission @ram-nad

ram-nad commented:
Thanks for this project sir. Indeed its very helpful. Is there any feature i could work on?

aamay001 commented on src/components/Resume/TechnicalSkills.js:
```diff
@@ -2,25 +2,28 @@ import React from 'react';
+              <div className="tech-skills-keyword" key={uuid()}>
+                <div className="keyword-name">{kw.name}</div>
+                <Stars lev={kw.level} />
+                {console.log(kw)}
```
Please remove the `console.log`

aamay001 commented on src/components/Resume/Stars.js:
```diff
@@ -0,0 +1,28 @@
+        </div>
+    );
+}
+
```
Missing propTypes for Starts.

aamay001 replied:
Yes, `propTypes` are included to add a small layer of type safety when passing props. Additionally, we can enforce certain props to be required. Here is the documentation for how to use: https://reactjs.org/docs/typechecking-with-proptypes.html

Let me know if you need further help with this.

aamay001 replied:
> Actually I haven't use propTypes before. Can you please help me in that.

Yes, propTypes are included to add a small layer of type safety when passing props. Additionally, we can enforce certain props to be required. Here is the documentation for how to use: https://reactjs.org/docs/typechecking-with-proptypes.html

Let me know if you need further help with this.

ram-nad commented:
Actually I haven't use propTypes before. Can you please help me in that.

aamay001 commented on src/components/Resume/Stars.js:
```diff
@@ -0,0 +1,28 @@
+    );
+}
+
+function level (lev) {
```
Please follow the `eslint` coding style from the project. Spacing rules and prefer arrow functions.

ram-nad commented:
I did all the suggested changes

ram-nad commented:
sir is there a plugin to do linting automatically according to the guidelines. Also i will work on making this feature optional but user will still have to enter their skills as object of format {name: 'Js'}.

aamay001 commented:
> sir is there a plugin to do linting automatically according to the guidelines. Also i will work on making this feature optional but user will still have to enter their skills as object of format {name: 'Js'}.

That's okay to use object format but we should support both the old format and new format to make the app backward compatible.  There are likely users that still have their resume saved in the old format. 

I use Visual Studio Code. Once you run 'npn install', VS Code will assert the rules. You may need to install this extension: https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint

ram-nad commented:
I am already using this extension. Currently working on the feature. Will get back as soon as it is done.

ram-nad commented:
- Made Skill Level Optional
- Added a disabled option for `ItemToogleButton`
- Added `env` variables in `eslintConfig` for making `document`, `window`, `it` etc available
- Updated `package.json`
- Added functions that will make `level` default as 4 if user is still using `string` in keywords

I have tried to fix all `eslint` errors that I could. I didn't have any understanding of `serviceWorker.js` so I could make modifications to it. If there is any other changes I would be happy to work on it.

ram-nad commented:
Fixed all `eslint` errors except for those in `serviceWorker.js`. That file has used `console.log` in many places which is shown as error in `eslint`. Also there are some errors like "`checkValidServiceWorker` was used before it was defined". I am clueless regarding how to solve it. Please help.

ram-nad commented:
I was also considering if we should add icons for `github`, `linkedin`, `email`, `phone` and `website`?

aamay001 commented:
> I was also considering if we should add icons for `github`, `linkedin`, `email`, `phone` and `website`?

If you would like to add this as a feature, please do so in a new PR. Thanks for your contributions!

aamay001 commented:
@ram-nad your changes are now live @ https://resumejs.netlify.com

aamay001 commented:
> Fixed all `eslint` errors except for those in `serviceWorker.js`. That file has used `console.log` in many places which is shown as error in `eslint`. Also there are some errors like "`checkValidServiceWorker` was used before it was defined". I am clueless regarding how to solve it. Please help.

Only need to resolve `eslint` errors in code that you submit. The Service Worker is a module included by `create-react-app`.
